### DISCUSSION TITLE
SU2_DEF functionality and mesh deformation of wing part only in an aircraft mesh

### DISCUSSION THREAD

**Discussion by Rashmikant15** [su2code/SU2#2405] [General]
Hello, SU2 community,
Thank you for being so supportive. I want to deform the wing part mesh in an aircraft mesh using SU2_DEF what should be its configuration file and is it necessary to identify the marker (MARKER_MOVING=  (wing_surface0) or we can do using fluidgrid.dat (grid point of wing mesh only)
Does SU2_DEF support RBF interpolation and deform an aircraft's mesh (wing part only) using the structural displacement file, or should I interpolate from my end and then apply fluid displacement for mesh deformation?

**Comment by bigfooted** [comment]
I always use the method as described here:
https://su2code.github.io/tutorials/Species_Transport/

You add the FFD box to the SU2 file, and then using DV_VALUE you can move the nodes of the FFD box in the x,y,z direction.  If you have a box of 5x5x5 = 125 nodes, then you 125*3=375 degrees of freedom and therefore 375 DV_VALUEs.

We have FFD_BLENDING= BEZIER as default, or alternatively FFD_BLENDING=BSPLINE_UNIFORM

DV_MARKER tells you which markers within the FFD box are allowed to move...

Maybe somebody else can comment on what you can and cannot do with regarding structural deformation.

**Reply by Rashmikant15** [reply]
Thank you so much, @bigfooted. I wanted to perform coupling, i.e., FSI analysis, so I want to clarify that this DEF function supports the deformation from structural nodes using RBF or performs only 3-D deformation post-RBF interpolation by itself.

**Reply by bigfooted** [reply]
Maybe @pcarruscag can help you with that question.

**Reply by pcarruscag** [reply]
SU2_DEF is not used for FSI, there are several FSI examples in the TestCases folder and repository.

**Reply by Rashmikant15** [reply]
Thank you so much @bigfooted @pcarruscag for clarifying the doubts. I want to use the SU2_DEF function to deform the 3D mesh (undeformed su2 mesh file) with the surface grid file (surface.dat) of the deformed surface. What should the configuration file and command code be? My question is similar to the "SU2 mesh deformation for steady state simulation #661" but I ran using the similar .cfg file and surface.dat but got error.

**Reply by pcarruscag** [reply]
You use DV_KIND=SURFACE_FILE and DV_FILENAME to point to the file you want.

**Reply by Rashmikant15** [reply]
Thank you so much @bigfooted @pcarruscag, for your continuous support. I used the method you explained and even looked at other threads but couldn't succeed so I need your support. I ran the code with the command "$ SU2_DEF wing.cfg" but still I got the error. 
Can you please check my files and cfg files too whether the details are correct or not? Is the format of the *.dat file correct?

Once again thank you so much for your help. 
[SU2_DEF_test.zip](https://github.com/user-attachments/files/18368542/SU2_DEF_test.zip)

**Reply by pcarruscag** [reply]
What error do you get?

**Reply by pcarruscag** [reply]
I don't see it here now but I got a reply to my email with errors about deprecated options, you probably based your config on something old, just follow the advice on screen by checking what we have in config_template or if the suggested alternatives seem correct use them

**Reply by Rashmikant15** [reply]
@pcarruscag Thank you so much for your support and so sorry I deleted the previous comment as I succeeded in executing the SU2_DEF for mesh deformation. I have a concern about the no deformed surface visualization file for verification of deformation. Although the surface_deformed.vtu file is generated, this is not working in paraview. Does SU2 only provide the volume_deformed.vtu which is working well? I want to verify whether the deformation is occurring or not. 

Once again thank you so much for your support. 
![Screenshot 2025-01-10 124700](https://github.com/user-attachments/assets/66b3019c-88dc-47dd-9a56-a5d3fd32e95e)
[comp_wing_DEF.zip](https://github.com/user-attachments/files/18370451/comp_wing_DEF.zip)
